### PR TITLE
prometheus: Document new behavior of unavailable or unknown entities

### PR DESCRIPTION
**Pull request by agoode**
## Proposed change

This documents the new prometheus behavior added in https://github.com/home-assistant/core/pull/125492.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/125492
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Introduced a new configuration option, `export_unavailable_metrics`, allowing users to control metric exports during entity unavailability. The default value will change from `true` to `false` in early 2025.

- **Documentation**
	- Enhanced documentation for the `prometheus` integration, clarifying the behavior of metrics in unavailable or unknown states.
	- Updated the explanation for the `entity_available` metric and removed the need for the `unless` clause in queries.
	- Expanded configuration examples for both basic and advanced setups, including guidance on filtering specific domains and entities.

### CODE REVIEW

**Review by knyar — APPROVED** [review]

**Review by knyar — APPROVED** [review]

File: source/_integrations/prometheus.markdown [hunk 1]
```diff
@@ -189,22 +189,28 @@ When looking into the metrics on the Prometheus side, there will be:
...
 
-When the Prometheus exporter starts (typically when Home Assistant starts), all non-excluded entities in an unavailable or unknown state are not be exported until they are available again. If the entity goes into state unavailable or unknown again, the value exported will always be the latest known one.
+When the Prometheus exporter starts (typically when Home Assistant starts), all non-excluded entities in an unavailable or unknown state are not exported until they are available and known.
+
+If an available entity goes into state unavailable or unknown, then it will automatically be unexported and return again automatically when available and known.
 
-While an entity is in those states, the `entity_available` corresponding metric is set to 0. This metric can be used to filter out values while the entity is unavailable or in an unknown state thanks to a [recording rule](https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/).
+{% note %}
+Prior to Home Assistant 2024.11, this component would continue to export the latest known value when an entity became unavailable or unknown, and set its corresponding `entity_available` metric to 0.
```
Comment (c0ffeeca7) [inline source/_integrations/prometheus.markdown]: normally, we only document current behavior. Do you think this could be removed?

Comment (c0ffeeca7) [inline source/_integrations/prometheus.markdown]:
```suggestion
```

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @agoode :+1:

**Comment by agoode** [comment]
Thank you for the edit!

**Pull request merged**